crates/uv/src/commands/python/install.rs

```diff
@@ -1032,20 +1031,33 @@ fn find_matching_bin_link<'a>(
     mut installations: impl Iterator<Item = &'a ManagedPythonInstallation>,
     path: &Path,
 ) -> Option<&'a ManagedPythonInstallation> {
-    let target = if cfg!(unix) {
-        if !path.is_symlink() {
-            return None;
-        }
-        fs_err::canonicalize(path).ok()?
-    } else if cfg!(windows) {
-        let launcher = Launcher::try_from_path(path).ok()??;
-        if !matches!(launcher.kind, LauncherKind::Python) {
+    #[cfg(not(any(unix, windows)))]
+    {
+        unreachable!("Only Windows and Unix are supported");
+    }
+
+    #[cfg(unix)]
+    {
+        let target = if !path.is_symlink() {
             return None;
-        }
-        dunce::canonicalize(launcher.python_path).ok()?
-    } else {
-        unreachable!("Only Windows and Unix are supported")
-    };
+        } else {
+            fs_err::canonicalize(path).ok()?
+        };
```

Comment:
nit: you can just do this check and return and then after do `let target = fs_err::...`